### PR TITLE
fix(react-grid): add type checking for next column in getNextColumnName

### LINKED ISSUE #2346: TypeError: Cannot read property 'name' of undefined adter update to 2.0.5
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...
- [x] React Grid

### Current Behaviour
After updating devexpress dependencies (devexpress/dx-react-core, devexpress/dx-react-grid and dx-react-grid-material-ui) from 2.0.4 to 2.0.5 the following probem exists in node_modules/@devexpress/dx-grid-core/dist/dx-grid-core.es.js:

TypeError: Cannot read property 'name' of undefined

image

### Environment
- react: 16.9.0
- browser: Chrome
- bootstrap: none
- react-bootstrap: none
- material-ui: 4.4.2

### PR DESCRIPTION
**Pull request by LazyLahtak**
Fixes #2346

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/table-header-row/helpers.ts [hunk 1]
```diff
@@ -71,7 +71,8 @@ export const getNextColumnName: NextColumnNameFn = (tableColumns, columnName) =>
   const index = tableColumns.findIndex(elem =>
       elem.type === TABLE_DATA_TYPE && elem.column!.name === columnName,
     );
-  return index >= 0 && index < tableColumns.length - 1
+  const haveNextColumn = index >= 0 && index < tableColumns.length - 1;
+  return haveNextColumn && tableColumns[index + 1].type === TABLE_DATA_TYPE
```
Comment (ushkal) [inline packages/dx-grid-core/src/plugins/table-header-row/helpers.ts]: Let's move it to a function

**Comment by LazyLahtak** [comment]
FPR

File: packages/dx-grid-core/src/plugins/table-header-row/helpers.ts [hunk 2]
```diff
@@ -67,11 +68,16 @@ export const getLastColumnName: LastColumnNameFn = (tableColumns) => {
     : undefined;
 };
 
-export const getNextColumnName: NextColumnNameFn = (tableColumns, columnName) => {
+const nextColumnName: NextColumnNameFn = (tableColumns , index) => {
+  const haveNextColumn = index >= 0 && index < tableColumns.length - 1;
```
Comment (MaximKudriavtsev) [inline packages/dx-grid-core/src/plugins/table-header-row/helpers.ts]: I think the next changes are more cleary

`haveNextColumn` -> `isNextColumnExists`

OR

```js
const nextColumn = index >= 0 && index < tableColumns.length - 1 && tableColumns[index + 1];
return nextColumn.type === TABLE_DATA_TYPE
  ? nextColumn.name
  : undefined;
```

**Comment by LazyLahtak** [comment]
FPR

**Pull request merged**